flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/enumerator/IcebergEnumeratorPosition.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink.source.enumerator;
+
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.base.Objects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+class IcebergEnumeratorPosition {
+  private final Long snapshotId;
+  private final Long snapshotTimestampMs;
+
+  IcebergEnumeratorPosition(Long snapshotId, Long snapshotTimestampMs) {
+    // either both are null or both are not null
+    Preconditions.checkArgument((snapshotId == null && snapshotTimestampMs == null) ||
+            (snapshotId != null && snapshotTimestampMs != null),
+        "snapshot id and timestamp should be either both null or both not null");
+    this.snapshotId = snapshotId;
+    this.snapshotTimestampMs = snapshotTimestampMs;
+  }
+
+  public Long snapshotId() {
+    return snapshotId;
+  }
+
+  public Long snapshotTimestampMs() {
```

Comment:
If this is just for logging, we may want to use `toString` so we don't expose this. I'm concerned about people trying to use it later to filter.

Reply:
realized this getter is also used in unit test. Since this class is non public, I will mark the two getters methods as package default scope too.